src/modules/ethereum/account-abstraction/fetchAndAssignPaymaster.ts

```diff
@@ -96,6 +97,7 @@ export async function fetchAndAssignPaymaster<T extends IncomingTransaction>(
     apiClient,
   });
   if (eligible && paymasterParams) {
+    txCopy.type = zksUtils.EIP712_TX_TYPE;
```

Comment:
I wonder why we didn't have this type set to EIP712_TX_TYPE before...

Reply:
The types are derived by ethers library from tx fields